### PR TITLE
AYR-1013 - Re-write table HTML

### PR DESCRIPTION
**Pull request by ltrasca**
## Changes in this PR

- Added generic table BEM styles that can be re-used across the application when building tables
- Replaced most table element classes with the ones mentioned above
- Changed HTML templates to only render one table row per iterated object (as opposed to having one for mobile and one for desktop). This single row uses some of the classes mentioned above to hide/show content depending on the width of the device.
- Changed the structure of HTML table column headers for some tables - these had used the same method of having duplicate elements for mobile and desktop.
- Adjusted some margins, gaps and column widths to be consistent across tables
- Changed and slightly refactored unit tests & utility functions for the pages included in the ticket to not use classes when looking for specific elements, instead it selects them based on the element tag and gets rid of invisible elements before-hand.
- On the /browse/consignment/<id> page, on mobile, changed the header for date of record and file name to be "Date of record / File name", previously this was just "Date of record" even though the column also included the file name

The changes for this ticket apply to these pages:

```
/browse
/browse/transferring_body
/browse/series
/browse/consignment
/search/transferring_body
```

## JIRA ticket
https://national-archives.atlassian.net/browse/AYR-1013

## Screenshots of UI changes

Attached in comments 👇

- [ ] Requires env variable(s) to be updated

### CODE REVIEW

**Comment by ltrasca** [comment]
<details>
  <summary>
    Click to view screenshots
  </summary>
Before

Desktop

image
image
image
image
image

Mobile
image
image
image
image
image



After

Desktop

image
image
image
image
image

Mobile

image
image
image
image
image
</details>

File: e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression_mobile/webkit-browse_transferring_body_mobile.png
Comment (colinbowen) [inline e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression_mobile/webkit-browse_transferring_body_mobile.png:1]: Might want to check this one with @Terry-Price. I think having the cell data aligned to the bottom row was a UI decision made during development.

File: e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression_mobile/webkit-browse_consignment_mobile.png
Comment (colinbowen) [inline e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression_mobile/webkit-browse_consignment_mobile.png:1]: The altered spacing doos look better here, good to confirm with @Terry-Price

File: e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression/webkit-search_transferring_body.png
Comment (colinbowen) [inline e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression/webkit-search_transferring_body.png:1]: File name width may have to be checked to ensure it allows fow filenames with a large amount of characters.

**Review by colinbowen — COMMENTED** [review]
Nice PR. I think we might need to check some changes in the table regarding spacing & the text change with Terry but it's much cleaner :slightly_smiling_face:

File: e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression/webkit-search_transferring_body.png
Comment (ltrasca) [reply e2e_tests/snapshots/test_visual_regression/test_css_no_visual_regression/webkit-search_transferring_body.png:1]: with slightly longer file names the table adjusts in width it seems, we could also set more fixed-ish widths to this (perhaps 30% or something along those lines)
image

**Comment by Terry-Price** [comment]
A few extra amends with column widths I've run through with Lucian and we are there.

**Review by Terry-Price — APPROVED** [review]
ran through a few minor spacing improvements and these are looking good.

**Review by anthonyhashemi — APPROVED** [review]
Looks good, the variable width css is clever, Ive never seen it before 😎

**Pull request merged**